### ISSUE TITLE
Consider switching libnacl with PyNaCl

### ISSUE THREAD

**Issue by uio-torjus** [unioslo/tsd-file-api#207]
Currently libnacl does not work with the newest version of libsodium.

PyNaCl seems to be better maintained than libnacl. We should consider switching.

**Comment by haatveit** [comment]
Agreed, but last time I checked on this it seemed like PyNaCl doesn't provide an interface for libsodium's `crypto_stream_xor()` function, which we use extensively. Would probably have to try PR-ing support for this upstream before we can switch over, unless I'm mistaken.

**Comment by leondutoit** [comment]
Or PR supporting the newest version of libsodium in libnacl? Wonder which one is more effort.

**Comment by leondutoit** [comment]
@uio-torjus now that your PR was merged, can we use the newest version of libsodium?
